src/utils/createClient.ts

```diff
@@ -1,15 +1,16 @@
-import {Role, SpvWalletClientExtended} from '@/contexts/SpvWalletContext.tsx';
+import { SpvWalletClientExtended } from '@/contexts/SpvWalletContext.tsx';
 import { AccessKeyWithSigning, AdminKey, SpvWalletClient, XprivWithSigning } from '@bsv/spv-wallet-js-client';
-import logger from '@/logger/intex.ts';
+import { Role, TRole } from '@/contexts/AuthContext.tsx';
+import logger from '@/logger';
 
-export const createClient = async (role: Role, key: string) => {
+export const createClient = async (role: TRole, key: string) => {
   const serverUrl = window.localStorage.getItem('login.serverUrl') ?? '';
 
   let clientOptions: any = {};
```

Comment:
You mean create a client here instead of clientOptions ?

`if (role === Role.Admin && key.startsWith('xprv')) {
    clientOptions = { adminKey: key } as AdminKey;
  } else if (role === Role.User) {
    if (key.startsWith('xprv')) {
      clientOptions = { xPriv: key } as XprivWithSigning;
    } else {
      clientOptions = { accessKey: key } as AccessKeyWithSigning;
    }
  }`

Reply:
Changed